### DISCUSSION TITLE
How to handle predictions for (possibly multiple) future time points?

### DISCUSSION THREAD

**Discussion by samuell** [featureform/featureform#441] [Q&A]
We have a lot of use cases involving time series-based predictions, which brought up some questions regarding how to best implement that in featureform. The first comes below.

Specifically we are wondering about how it would look like when calling the serving client for various time steps into the future?
E.g. say that we want to make a prediction value for each five minutes in the coming hour.
These time points will not exist in the dataset already, so it won't make sense to ask for the output of a specific our.

The way I can see this be done is to add the multiple time "horizons" as separate labels, such as `predicted_value_in_5m`, `predicted_value_in_10m`, and so forth.

One thing I'm getting worried about here though is how to make sure that you know exactly which time point (in the future) a specific prediction applies for. E.g. how can you know that the input data sources are updated, so that "in 5 minutes" is really the next five minutes, and not 4, 3, 2 or 1 minute from now ... since the serving API doesn't let you send in a specific time point as the basis for the predicted value?

If I understand it correctly, in Feast you can solve this by sending in a "template dataframe" with a number of time points for which you want to predict values for, and it will fill in the requested columns for those time points. 

But was wondering how to solve this in FeatureForm?

**Comment by architchandra** [comment]
I got mentioned in a lewd-seeming comment on this thread. I've never even used this tool. What is going on? Any idea about how the poster got these particular usernames? I hope this is not some shady marketing tactic.

Here's a screenshot of the email notification I received.

![Screenshot 2022-12-29 at 10 31 14 AM](https://user-images.githubusercontent.com/8830075/209905771-1ef5556c-40db-4550-8104-43a0bc3c35c5.png)

**Reply by samuell** [reply]
Ouch, scammers found their way to GitHub ... doens't feel like the most receptible crowd though :thinking:
